src/figuresClasses.ts

```diff
@@ -1,19 +1,95 @@
-export interface Figure {
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+enum Color {
+  Red ='red',
+  Green = 'green',
+  Blue = 'blue',
+}
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Figure['shape'] = Shape.Triangle;
+
+  constructor(
+    public color: Figure['color'],
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('The side is less or equal to zero');
+    }
 
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('Two sides are less or equal to third side');
+    }
+  }
+
+  getArea(): number {
+    const semiperimeter = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(
+      semiperimeter
+      * (semiperimeter - this.a)
+      * (semiperimeter - this.b)
+      * (semiperimeter - this.c),
+    );
+
+    return area;
+  }
 }
 
-export class Circle {
+export class Circle implements Figure {
+  shape: Figure['shape'] = Shape.Circle;
+
+  constructor(
+    public color: Figure['color'],
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('the radius is less or equal to zero');
+    }
+  }
 
+  getArea(): number {
+    const area = Math.PI * this.radius * this.radius;
+
+    return Math.floor(area * 100) / 100;
+  }
 }
 
 export class Rectangle {
+  shape: Figure['shape'] = Shape.Rectangle;
+
+  constructor(
+    public color: Figure['color'],
+    public width: number,
+    public height: number,
+  ) {
+    if (height <= 0 || width <= 0) {
+      throw new Error('Height or width are less or equal to zero');
+    }
+  }
 
+  getArea(): number {
+    return this.width * this.height;
+  }
 }
 
-export function getInfo(figure) {
+export function getInfo(figure: Figure): string {
+  const area = figure.getArea();
+  const { color, shape } = figure;
+  const formattedArea = area % 1 !== 0
+    ? area.toFixed(2)
+    : area;
```

Comment:
You don't need do this here
You can reuse your function in every `getArea` method in every class(check my comment above about creating function for rounding)